websites/S/Scratch/presence.ts

```diff
@@ -239,23 +230,69 @@ presence.on("UpdateData", () => {
 			presenceData.state = "Scratch Foundation";
 		}
 	}
-	if (window.location.hostname.toLowerCase().includes("scratch-wiki.info")) {
-		presenceData.details = "Viewing page:";
-		presenceData.state = "Wiki";
+	if (window.location.href.toLowerCase().includes("scratch-wiki.info")) {
+		if (window.location.href.toLowerCase().includes("en.scratch-wiki.info")) {
+			presenceData.details = "Viewing page:";
+			presenceData.state = "English Scratch Wiki";
+		} else if (
+			window.location.href.toLowerCase().includes("de.scratch-wiki.info")
+		) {
+			presenceData.details = "Viewing page:";
+			presenceData.state = "German Scratch Wiki";
+		} else if (
+			window.location.href.toLowerCase().includes("ru.scratch-wiki.info/")
+		) {
+			presenceData.details = "Viewing page:";
+			presenceData.state = "Russian Scratch Wiki";
+		} else if (
+			window.location.href.toLowerCase().includes("nl.scratch-wiki.info/")
+		) {
+			presenceData.details = "Viewing page:";
+			presenceData.state = "Dutch Scratch Wiki";
+		} else if (
+			window.location.href.toLowerCase().includes("id.scratch-wiki.info/")
+		) {
+			presenceData.details = "Viewing page:";
+			presenceData.state = "Indonesian Scratch Wiki";
+		} else if (
+			window.location.href.toLowerCase().includes("ja.scratch-wiki.info/")
+		) {
+			presenceData.details = "Viewing page:";
+			presenceData.state = "Japanese Scratch Wiki";
+		} else if (
+			window.location.href.toLowerCase().includes("hu.scratch-wiki.info/")
+		) {
+			presenceData.details = "Viewing page:";
+			presenceData.state = "Hungarian Scratch Wiki";
+		} else if (
+			window.location.href.toLowerCase().includes("fr.scratch-wiki.info/")
+		) {
+			presenceData.details = "Viewing page:";
+			presenceData.state = "French Scratch Wiki";
+		} else if (
+			window.location.href.toLowerCase().includes("test.scratch-wiki.info/")
+		) {
+			presenceData.details = "Viewing page:";
+			presenceData.state = "Test Scratch Wiki";
+		} else {
+			presenceData.details = "Viewing page:";
+			presenceData.state = "Scratch Wiki";
+		}
 	}
 	if (window.location.hostname.toLowerCase().includes("scratchjr.org")) {
 		presenceData.details = "Viewing page:";
 		presenceData.state = "Scratch JR";
 	}
 	if (
-		window.location.hostname.toLowerCase().includes("donationpay.org") &&
-		window.location.pathname.toLowerCase().includes("/scratchfoundation")
+		window.location.href
+			.toLowerCase()
+			.includes("donationpay.org/scratchfoundation")
 	) {
 		presenceData.details = "Viewing page:";
 		presenceData.state = "Donate";
 	}
 	if (
-		window.location.hostname.toLowerCase().includes("scratched.gse.harvard.edu")
+		window.location.href.toLowerCase().includes("scratched.gse.harvard.edu")
```

Comment:
I don't see why you should use href here

Reply:
i did some testing, it seems that href manages to detect url changes better